airflow-core/src/airflow/logging_config.py

```
@@ -19,40 +19,65 @@
-    logging_class_path = ""
-    try:
-        logging_class_path = conf.get("logging", "logging_config_class")
-    except AirflowConfigException:
-        log.debug("Could not find key logging_config_class in config")
+    global REMOTE_TASK_LOG
+    logging_class_path = conf.get("logging", "logging_config_class", fallback=None)
 
-    if logging_class_path:
-        try:
-            logging_config = import_string(logging_class_path)
+    user_defined = logging_class_path is not None
 
-            # Make sure that the variable is in scope
-            if not isinstance(logging_config, dict):
-                raise ValueError("Logging Config should be of dict type")
+    logging_class_path = (
+        logging_class_path or "airflow.config_templates.airflow_local_settings.DEFAULT_LOGGING_CONFIG"
+    )
+    try:
+        logging_config = import_string(logging_class_path)
 
+        # Make sure that the variable is in scope
+        if not isinstance(logging_config, dict):
+            raise ValueError("Logging Config should be of dict type")
+
+        if user_defined:
             log.info("Successfully imported user-defined logging config from %s", logging_class_path)
-        except Exception as err:
-            # Import default logging configurations.
-            raise ImportError(f"Unable to load custom logging from {logging_class_path} due to {err}")
+
+    except Exception as err:
+        # Import default logging configurations.
+        raise ImportError(
+            f"Unable to load {'custom ' if user_defined else ''} logging config from {logging_class_path} due to {err}"
+        )
     else:
-        logging_class_path = "airflow.config_templates.airflow_local_settings.DEFAULT_LOGGING_CONFIG"
-        logging_config = import_string(logging_class_path)
-        log.debug("Unable to load custom logging, using default config instead")
+        mod = logging_class_path.rsplit(".", 1)[0]
+        with suppress(ImportError):
```

Remove suppress?

---

Will the flipside to removing this be that we cannot use airflow normally due to some service crashing?

A better thing would be is to log a warning that logs are not going to work as expected than fail silently?

---

Yeah, I've logged it instead, but not failing is better I think, that matches the behavoiur with failing to import custom logging config.